x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -435,10 +435,16 @@ else if (plan instanceof OrderBy) {
                 }
             }
 
-            return result;
+            return filterUnsupportedProjections(result);
         }
 
-        static List<NamedExpression> expandStar(UnresolvedStar us, List<Attribute> output) {
+        private List<NamedExpression> filterUnsupportedProjections(List<NamedExpression> projections) {
+            return projections.stream()
+                .filter(p -> p.resolved() == false || isTypeSupportedInVersion(p.dataType(), configuration.version()))
```

Comment:
Why `||`? I would have thought `&&` is the one...

Reply:
I think that if the fields is unresolved the type might not be available.